### PR TITLE
Fix 4755

### PR DESCRIPTION
**Pull request by belforte**
do not use env -i before gfal-* but simply undo scram environment as a general protection against CMSSW vs. GRID conflicts
Also  a bit of code cleanup for readibility

### CODE REVIEW

File: src/python/CRABClient/Commands/remote_copy.py [hunk 1]
```diff
@@ -140,10 +140,13 @@ def __call__(self):
                 continue
 
             ##### Creating the command
+            # better to execut grid commands in the pre-CMS environment
+            undoScram = "which scram >/dev/null 2>&1 && eval `scram unsetenv -sh`"
+            cmd = undoScram + cmd
```
Comment (lecriste) [inline src/python/CRABClient/Commands/remote_copy.py]: `cmd` is not defined before, any reason to add `+ cmd`?
Anyhow it is overridden by https://github.com/dmwm/CRABClient/pull/4758/files#diff-62f68c1aa854c51b75d7304faaff2af7R153 and not used before.

Comment (belforte) [reply src/python/CRABClient/Commands/remote_copy.py]: thanks @lecriste  for spotting this. It was a plain mistake.  But I do not understand what you write about it being overridden. I do not understand if I should fix this or you did somethig already, or ... ?

Comment (lecriste) [reply src/python/CRABClient/Commands/remote_copy.py]: `cmd` is assigned in L145 and reassigned in L153 without being used before, so the former value is never used.
I did not touch the code.

Comment (belforte) [reply src/python/CRABClient/Commands/remote_copy.py]: thanks, understood. never the less L145 will cause a failure ! I will fix this.

Comment (belforte) [reply src/python/CRABClient/Commands/remote_copy.py]: also the way is is coded now will plainly not work !

**Comment by belforte** [comment]
@lecriste 
this was tested at DESY as well and it solves the problem #4755 . I propose to merge,tag and deploy in preprod toghehter with the new WMCore tag.

**Comment by lecriste** [comment]
+1
Shall I merge and tag or do you want to do it?
To go in preprod the tag shoul be be 3.3.1805.rc2

**Comment by belforte** [comment]
please do it. I would most likely get confused with server and mess things up :-(

**Pull request merged**